src/main/java/ch/njol/skript/lang/parser/ParserInstance.java

```diff
@@ -116,8 +115,34 @@ public SkriptEvent getCurrentSkriptEvent() {
 	public List<TriggerSection> getCurrentSections() {
 		return currentSections;
 	}
-	
-	public List<Loop> getCurrentLoops() {
+
+	/**
+	 * @return whether {@link #getCurrentSections()} contains
+	 * an section instance of the given class (or subclass).
+	 */
+	public boolean isCurrentSection(Class<? extends TriggerSection> sectionClass) {
+		for (TriggerSection triggerSection : currentSections) {
+			if (sectionClass.isInstance(triggerSection))
+				return true;
+		}
+		return false;
+	}
+
+	/**
+	 * @return the outermost section which is an instance of the given class.
+	 * Returns {@code null} if {@link #isCurrentSection(Class)} returns {@code false}.
+	 */
+	@SuppressWarnings("unchecked")
+	@Nullable
+	public <T extends TriggerSection> T getCurrentSection(Class<T> sectionClass) {
+		for (TriggerSection triggerSection : currentSections) {
+			if (sectionClass.isInstance(triggerSection))
+				return (T) triggerSection;
+		}
+		return null;
+	}
+
+	public List<LoopSection> getCurrentLoops() {
```

Comment:
Why would there need to be this method when we could just use `getCurrentSection(LoopSection.class)`?
I would deprecate the method, and invoke that method instead to not break addons immediately.

Reply:
I removed the method, mainly because it is only present in the first alpha, so not many addons will have seen it, and I don't think any addons would need the method in the first place.